cluster-autoscaler/cloudprovider/oci/README.md

```diff
@@ -24,14 +24,30 @@ We recommend setting up and configuring the Cluster Autoscaler to use
 [Instance Principals](https://docs.oracle.com/en-us/iaas/Content/Identity/Tasks/callingservicesfrominstances.htm)
 to authenticate to the OCI APIs.
 
-The following policy provides the minimum privileges necessary for Cluster Autoscaler to run:
+The following policy provides the privileges necessary for Cluster Autoscaler to run:
 
 1: Create a compartment-level dynamic group containing the nodes (compute instances) in the cluster:
 
 ```
 All {instance.compartment.id = 'ocid1.compartment.oc1..aaaaaaaa7ey4sg3a6b5wnv5hlkjlkjadslkfjalskfjalsadfadsf'}
 ```
 
+**WARNING:** The above dynamic group with its tenancy-level policy in
```

Comment:
Thanks for catching this. 

Actually, I do not believe the requirement for a tenancy-level policy is accurate. See [here](https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengusingclusterautoscaler_topic-Working_with_Cluster_Autoscaler_as_Cluster_Add-on.htm). 

So, rather than a security warning, we should update the steps for the policy instructions to something like the following:


2: Create a policy to allow nodes to manage node-pools and/or instance-pools:

```
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to manage instance-family in compartment <compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to manage volume-family in compartment <compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to use subnets in compartment <compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to read virtual-network-family in compartment <compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to use vnics in compartment <compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to inspect compartments in compartment <compartment-name>
# if using node pools
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to manage cluster-node-pools in compartment <compartment-name>
# if using instance pools
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to manage instance-pools in compartment <compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to manage instance-configurations in compartment <compartment-name>
```


**Note:** If  a node or instance pool belongs to one compartment, and the network resources used by the node pool belong to a different compartment, you have to create policies in both compartments as follows:

a: In the node or instance pool's compartment, create a policy with policy statements:

```
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to manage instance-family in compartment <nodepool-compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to use subnets in compartment <nodepool-compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to use vnics in compartment <nodepool-compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to inspect compartments in compartment <nodepool-compartment-name>
# if using node pools
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to manage cluster-node-pools in compartment <nodepool-compartment-name>
# if using instance pools
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to manage instance-pools in compartment <nodepool-compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to manage instance-configurations in compartment <nodepool-compartment-name>
```

b. In the network resources' compartment, create a policy with policy statements:
```
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to use subnets in compartment <network-compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to read virtual-network-family in compartment <network-compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to use vnics in compartment <network-compartment-name>
Allow dynamic-group acme-oci-cluster-autoscaler-dyn-grp to inspect compartments in compartment <network-compartment-name>
```

Reply:
Hi @jlamillan , 

Thanks for your time reviewing my patch.

The documentation (https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengusingclusterautoscaler_topic-Working_with_Cluster_Autoscaler_as_Cluster_Add-on.htm) also has  the same issue I'd suggest. 

The policy defines what dynamic-group can do, and the dyanmic group defines who (which instances) can do with the policy:

[1] https://docs.oracle.com/en-us/iaas/Content/Identity/Tasks/managingdynamicgroups.htm

Dynamic group supports a few selectors, one of them is `instance.compartment.id` that will select all instances in that compartment. That means, all instances can gain all privileges attached via the dynamic group's policy. 

Please fix me if I'm wrong.

Thanks a lot.

Reply:
Thanks for the clarification. 

I agree it's worth adding a _note_ about using defined tags. However, the proposed WARNING seems a bit strong. Matching all instances in a compartment is not inherently insecure and may be just fine for many environments.

We also need to keep in mind that this provider supports both node-pools (OKE) and instance-pools based nodes. So, rather than trying to document how to configure defined tags on either a node-pool or an instance-pool here, how about we link to the relevant instructions?

i.e. update this:

> WARNING: The above dynamic group with its tenancy-level policy in the next step will allow all machines in the specified compartment to handle all important computing resources. This may lead to some serious security problems on your system. It's highly recommended to limit the scope of this setting by for example. 
> 1. Using a proper...
> 2. Setting up Defined-Tag...
> 3. Refining the dynamic group...

To something like:

> Note: the matching rule in the dynamic group above includes _all_ instances in the specified compartment. If this is too broad for your requirements, it can be updated to only match instances that have a certain defined-tag in the compartment. 

> Ensure defined tags are set on your [node-pool](https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengtaggingclusterresources_tagging-oke-resources_node-tags.htm) or [instance-pool](https://docs.oracle.com/en-us/iaas/Content/Compute/Tasks/creatinginstanceconfig.htm) nodes.  

> Then, update the matching rule on the dynamic group to match that tag:

```
All {instance.compartment.id = 'ocid1.compartment.oc1..aaaaaaaa7ey4sg3a6b5wnv5hlkjlkjadslkfjalskfjalsadfadsf', tag.MyTagNamespace.MyNodeRole = 'ClusterAutoscaler'}
```

Reply:
@icy did you wanna make the proposed change or would you like me to?

Reply:
yes @jlamillan , I am a bit busy with new laptop now and intend to update my patch tomorrow. Thanks for your patience.

Reply:
No rush.

Reply:
@jlamillan I've updated the patch. Please let me know if I need to change something. Thanks.